### PR TITLE
Add addon ID and version to the schema

### PR DESCRIPTION
**Pull request by feerrenrut**
- Some fixes to how how lint / tests are run with tox
- Updates the schema to include addon ID and addon version
- The addon version (from the manifest) must be read and compared to the file name and submitted data.
  - This will be required in NVDA as well. A common library should be considered.

With this change, the addon ID and version are checked to match the path for the submission as well as the addon manifest (extracted from the *.nvda-addon)

### CODE REVIEW

**Comment by feerrenrut** [comment]
In https://github.com/nvaccess/addon-store-submission/pull/16 a case where the addon version from the manifest is used a version name rather than a version number.

File: tox.ini [hunk 1]
```diff
@@ -9,12 +9,16 @@ envlist = py
 
 [testenv]
 deps =
-    mock
-    jsonschema==3.1
-    git+git://github.com/DiffSK/configobj@3e2f4cc#egg=configobj
+	mock
```
Comment (seanbudd) [inline tox.ini]: mock should also be removed here, to use unittest.mock instead.

File: tox.ini [hunk 2]
```diff
@@ -9,12 +9,16 @@ envlist = py
...
+	flake8==3.9.2
+	# flake8-tabs version 2.3.2 gives spurious errors:
+	# "ET113 (flake8-tabs) use of alignment as indentation, but option continuation-style=hanging does not permit this"
+	flake8-tabs==2.2.2
+	-r requirements.txt
 
 commands =
-    python -m unittest discover
+	python -m unittest discover
+	flake8
```
Comment (seanbudd) [inline tox.ini:20]: Just a note here to consider removing `use-pycodestyle-indent = False` to be consistent with the other discussion.

File: _tests/test_validate.py [hunk 3]
```diff
@@ -165,38 +169,160 @@ def test_checkDescriptionMatches_invalid(self):
...
+		- Submission file name '<addonID>/<version>.json'
+		- `addonVersionField` within the submission JSON data
+		"""
 		errors = list(
-			validate.checkVersionMatchesFilename(self.manifest, VALID_SUBMISSION_JSON_FILE)
+			validate.checkVersions(self.manifest, VALID_SUBMISSION_JSON_FILE, self.submissionData)
 		)
 		self.assertEqual(errors, [])
 
 	def test_checkVersionMatchesFilename_invalid(self):
```
Comment (seanbudd) [inline _tests/test_validate.py]: Should the function be renamed to something like the following? Just to be consistent.
```suggestion
	def test_checkVersionMatches_invalidFilename(self):
```

**Comment by feerrenrut** [comment]
Changes have been made as per https://github.com/nvaccess/addon-store-submission/pull/16#issuecomment-911518440
With the extension that the filename must be a fully qualified version string. EG `4.2.0.json` rather than `4.2.json`

File: _validate/validate.py [hunk 4]
```diff
@@ -30,22 +29,6 @@
 JsonObjT = typing.Dict[str, typing.Any]
 
 
-@enum.unique
-class ValidationErrorMessage(enum.Enum):
```
Comment (seanbudd) [inline _validate/validate.py:33]: I'm curious to know why this pattern was removed - it made the tests easier to read as well.

File: _tests/test_validate.py [hunk 5]
```diff
@@ -62,143 +64,449 @@ def test_checkDownloadUrlFormat_validExampleURL(self):
...
+		- Submission file name '<addonID>/<version>.json'
+		- `addonId` within the submission JSON data
+	"""
+	def setUp(self):
+		self.submissionData = getValidAddonSubmission()
+		self.manifest = getAddonManifest()
+
+	def tearDown(self):
+		self.submissionData = None
+		self.manifest = None
```
Comment (seanbudd) [inline _tests/test_validate.py:212]: Could this repeated setUp/tearDown code be inherited from a base TestCase across all these TestCases?

File: _validate/validate.py (hunk 4 above)
Comment (feerrenrut) [reply _validate/validate.py:33]: I found the opposite, especially when the error messages changed meaning. It meant referring back to the original all the time. By putting the error message in the test it documents what the final output will be for each of the situations.

File: _tests/test_validate.py (hunk 5 above)
Comment (feerrenrut) [reply _tests/test_validate.py:212]: Possibly. I'd prefer not inheritance for code reuse, the tests don't necessarily have the same dependencies. It's pretty minor repetition. On the plus side it keeps the behavior of a related group of tests local.

**Review by seanbudd — APPROVED** [review]

**Pull request merged**